internal/repository/user.go

```diff
@@ -0,0 +1,122 @@
+package repository
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/dilyara4949/flight-booking-api/internal/domain"
+	"github.com/google/uuid"
+)
+
+type userRepository struct {
```

Comment:
why repo isn't exposed?

Reply:
since  NewUserRepository returns interface, i won't need access to this struct

Reply:
linter will send a warn in this case